src/components/auth/Login.jsx

```diff
@@ -29,10 +39,15 @@ export const Login = () => {
       }
    }
 
+   useEffect(() => {
```

Comment:
The idea of a custom hook is to centralize the logic in one place, so that we can use the compose pattern, find a way to move the useEffect from the components (Login, Register) to the custom hook.